### PR TITLE
FED- prod githubvalidation bug

### PR DESCRIPTION
**Pull request by w8896699**
* Fix existing bug on prod about input keyword undefined by changing redux data structure.
* add namespace and quotasize into search query

### CODE REVIEW

File: web/src/components/common/UI/GithubUserValidation/AdaptedGithubUserDisplay.tsx [hunk 1]
```diff
@@ -42,45 +42,47 @@ const AdaptedGithubUserDisplay: React.FC<any> = (props) => {
...
     dispatchGithubUserKeywordInput,
     ...rest
   } = props;
 
   useEffect(() => {
-    if (userFieldInputEvent.value !== allPersona[index].inputKeyword) {
+    if (userFieldInputEvent.value !== allPersona[persona][position].inputKeyword) {
       const inputValue = userFieldInputEvent.value;
-      dispatchGithubUserKeywordInput({ index, inputValue });
+      dispatchGithubUserKeywordInput({ persona, position, inputValue });
```
Comment (patricksimonian) [inline web/src/components/common/UI/GithubUserValidation/AdaptedGithubUserDisplay.tsx:55]: still a terrible name

File: web/src/components/common/UI/GithubUserValidation/AdaptedGithubUserDisplay.tsx [hunk 2]
```diff
@@ -42,45 +42,47 @@ const AdaptedGithubUserDisplay: React.FC<any> = (props) => {
...
     }
-  }, [userFieldInputEvent, allPersona, dispatchGithubUserKeywordInput, index]);
+    // eslint-disable-next-line
```
Comment (patricksimonian) [inline web/src/components/common/UI/GithubUserValidation/AdaptedGithubUserDisplay.tsx:57]: this doesn't seem right.

File: web/src/redux/githubID/githubID.reducer.ts [hunk 3]
```diff
@@ -18,45 +23,80 @@ const GithubIDBaseState = {
...
 
+interface GithubIDUserStore {
+  type: string;
+  payload: {
+    persona: 'productOwner' | 'technicalLeads';
+    position: number;
+    data: object;
+  };
+}
+type Action = GithubUserInputAction & GithubIDUserBasicAction & GithubIDUserStore;
```
Comment (patricksimonian) [inline web/src/redux/githubID/githubID.reducer.ts:55]: Shouldn't this be `|` not `&`?

File: web/src/redux/githubID/githubID.reducer.ts [hunk 4]
```diff
@@ -18,45 +23,80 @@ const GithubIDBaseState = {
...
-      return newArrayForUserRequest;
+
+      return newStateForUserRequest;
 
     case GithubIDActionTypes.GITHUB_USER_EXISTS:
-      const newArrayForUserExist = [...state];
-      newArrayForUserExist[action.payload] = {
-        ...state[action.payload],
+      const { persona: userExistPersona, position: userExistPosition } = action.payload;
+      const newArrayForUserExist = { ...state };
```
Comment (patricksimonian) [inline web/src/redux/githubID/githubID.reducer.ts:72]:
```suggestion
      const newState = { ...state };
```

Comment (patricksimonian) [reply web/src/redux/githubID/githubID.reducer.ts:72]: be mindful of your variable naming. the object is not even an Array. *slaps wrist*

File: web/src/redux/githubID/githubID.reducer.ts (hunk 3 above)
Comment (w8896699) [reply web/src/redux/githubID/githubID.reducer.ts:55]: Its &

**Review by patricksimonian — APPROVED** [review]

**Pull request merged**